x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_action_types/email.tsx

```diff
@@ -0,0 +1,540 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment } from 'react';
+import {
+  EuiFieldText,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFieldNumber,
+  EuiFieldPassword,
+  EuiComboBox,
+  EuiTextArea,
+  EuiSwitch,
+  EuiFormRow,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  ActionTypeModel,
+  ActionConnectorFieldsProps,
+  ActionConnector,
+  ValidationResult,
+  ActionParamsProps,
+} from '../../../types';
+
+export function getActionType(): ActionTypeModel {
+  const mailformat = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  return {
+    id: '.email',
+    iconClass: 'email',
+    selectMessage: i18n.translate(
+      'xpack.triggersActionsUI.components.builtinActionTypes.emailAction.selectMessageText',
+      {
+        defaultMessage: 'Send email from your server.',
+      }
+    ),
+    validateConnector: (action: ActionConnector): ValidationResult => {
+      const validationResult = { errors: {} };
+      const errors = {
+        from: new Array<string>(),
+        service: new Array<string>(),
+        port: new Array<string>(),
+        host: new Array<string>(),
+        user: new Array<string>(),
+        password: new Array<string>(),
+      };
+      validationResult.errors = errors;
+      if (!action.config.from) {
+        errors.from.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredFromText',
+            {
+              defaultMessage: 'Sender is required.',
+            }
+          )
+        );
+      }
+      if (action.config.from && !action.config.from.match(mailformat)) {
+        errors.from.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.formatFromText',
+            {
+              defaultMessage: 'Sender is not a valid email address.',
+            }
+          )
+        );
+      }
+      if (!action.config.port && !action.config.service) {
+        errors.port.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredPortText',
+            {
+              defaultMessage: 'Port is required.',
+            }
+          )
+        );
+      }
+      if (!action.config.service && (!action.config.port || !action.config.host)) {
+        errors.service.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredServiceText',
+            {
+              defaultMessage: 'Service is required.',
+            }
+          )
+        );
+      }
+      if (!action.config.host && !action.config.service) {
+        errors.host.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.error.requiredHostText',
+            {
+              defaultMessage: 'Host is required.',
+            }
+          )
+        );
+      }
```

Comment:
Lets get back to this with the issue [54191](https://github.com/elastic/kibana/issues/54191), because for now, I removed Service field from the form and need to fix it in the create issue (error messages will also be properly changed